draft-axu-dnsop-catalog-zone-xfr-properties.md

```diff
@@ -94,36 +94,94 @@ Body text \[REPLACE\]
+ns1.primaries.ZONELABEL2.zones.$CATZ  0  IN AAAA 2001:db8:35::53
+ns1.primaries.ZONELABEL2.zones.$CATZ  0  IN TXT "keyname-for-ns1"
+ns2.primaries.ZONELABEL2.zones.$CATZ  0  IN AAAA 2001:db8:35::54
+ns2.primaries.ZONELABEL2.zones.$CATZ  0  IN TXT "keyname-for-ns2"
+~~~
 
 ### TLSA
 
-Body text \[REPLACE\]
+The primaries property, with or without the extra label, MAY also have one or more TLSA resource records. This will apply restrictions to which keys can be used to transfer the member zone(s) using DNS over TLS or DNS over QUIC. (is this correct?)
 
-## Allow Notify
+\[WRITE EXAMPLE\]
 
-Body text \[REPLACE\]
+## Notify
+
+This property MAY be used to define the DNS NOTIFY message sending behavior of the consumer for the target zone(s). It MAY contain resource records of type A, AAAA and TXT.
+
+The A and AAAA records list hosts that the consumer will send DNS NOTIFY messages to when it loads a new version of the target zone(s).
+
+If a record of type TXT is not found, the consumer MAY also send NOTIFYs according to its default behavior as defined by its configuration and its code. However, if a TXT record is found, this default behavior MUST be suppressed, and NOTIFYs are only sent to the hosts listed in A and AAAA records if any. The value of the TXT record doesn't matter, and thus the number of TXT records also does not matter.
+
+~~~ ascii-art
+notify.$CATZ  0                          IN A 192.0.2.49
+
+ZONELABEL3.zones.$CATZ  0                IN PTR example.org.
+notify.ZONELABEL3.zones.$CATZ  0         IN AAAA 2001:db8:35::53
+notify.ZONELABEL3.zones.$CATZ  0         IN TXT "no default notifies"
+
+ZONELABEL4.zones.$CATZ  0                IN PTR sub.example.org.
+notify.ZONELABEL4.zones.$CATZ  0         IN AAAA 2001:db8:35::54
+notify.ZONELABEL4.zones.$CATZ  0         IN TXT ""
+~~~
+
+If there are any RRs attached to the notify property that are not covered by this document, they SHOULD be ignored.
  
+## Allow Query
+
+The allow-query property MAY be used to define an access list of hosts or networks that are allowed to send queries for the target zone(s).
+The resource record type MUST be either APL or CNAME. The APL record {{!RFC3123 (APL RR)}} MAY be used to define the access-list directly, while the CNAME record MAY be used to refer to an access-list already defined elsewhere. The CNAME MUST point to a name that has an APL record.
+
+~~~ ascii-art
+ZONELABEL5.zones.$CATZ  0                IN PTR example.local.
+allow-query.ZONELABEL5.zones.$CATZ  0    IN APL 1:10.0.0.0/8 !1:0.0.0.0/0 !2:0:0:0:0:0:0:0:0/0
+~~~
+
+If there are RRs other than APL or CNAME attached to the allow-query property, and if an APL RR cannot be found or there is a CNAME that doesn't point to an APL, then the most restrictive access list possible SHOULD be assumed.
+
+QUESTION1: should we define a label convention for pre-defining access-lists in a CATZ? RFC9432 had the group property to do something like this, but it refers to resources outside DNS.
```

Comment:
Indeed. I like to have groups configured too like this. See also Libor's comment in Issue #5